src/adapter/src/coord/peek.rs

```diff
@@ -490,10 +490,14 @@ impl<S: Append + 'static> crate::coord::Coordinator<S> {
                 inverse.entry(*compute_instance).or_default().insert(*uuid);
             }
             for (compute_instance, uuids) in inverse {
-                self.controller
+                // It's possible that this compute instance no longer exists because it was dropped
+                // while the peek was in progress. In this case we ignore the error and move on
+                // because the dataflow no longer exists.
+                // TODO(jkosh44) Dropping a cluster should actively cancel all pending queries.
```

Comment:
FYI, I played around with this locally for a bit. It's very possible that this isn't possible because `psql` doesn't know how to handle query cancelled responses that weren't initiated by `psql`. In fact it may be against the pgwire protocol since no cancel request was sent in the first place.

Another option would be to send a notice to the query that it should try and cancel because the cluster no longer exists.

Reply:
I'm not sure how psql works under the covers here, but I would imagine that it's able to handle cancel query requests coming from anything, because they present as a regular error response. Was that not your experience? Also you showed me some postgres session txn timeout code earlier where the timeout would send a cancelled error, which is the same error as from a cancel request connection.

Reply:
Oh yeah that's a good point, I must  have been doing something wrong then. When I tried the query was just hanging. I'll open an issue after merging this to cancel outstanding queries.